src/allocator.rs

```diff
@@ -1082,7 +1082,9 @@ fn test_concat_launder_small_number() {
 
     // this "launders" the small number into actually being allocated on the
     // heap
-    let atom2 = a.new_concat(1, &[atom1]).expect("new_substr");
+    let atom2 = a
+        .new_concat(1, &[a.nil(), atom1, a.nil()])
```

Comment:
this is somewhat unrelated. it makes it more likely that this `concat` actually causes a heap allocation for the return value